src/promptflow/promptflow/_sdk/_submitter/utils.py

```diff
@@ -198,21 +203,43 @@ def resolve_connections(flow: Flow, client=None, connections_to_ignore=None) ->
         )
 
     @staticmethod
-    def resolve_connection_names_from_tool_meta(tools_meta: dict, flow_dag: dict,):
-        connection_names = set({})
-        tool_names = set({})
-        if tools_meta:
-            packages = tools_meta.get("package", {})
-            for key, pkg in packages.items():
-                inputs = pkg.get("inputs", {})
-                if "connection" in inputs and inputs["connection"].get("type", []) == ["object"]:
-                    tool_names.add(key)
-            nodes = flow_dag.get("nodes", [])
-            for node in nodes:
-                if node.get("source", {}).get("tool", "") in tool_names:
-                    connection_names.add(node["inputs"]["connection"])
-            return list(connection_names)
-        return []
+    def resolve_used_connections(
+        flow: ProtectedFlow, tools_meta: dict, client=None, connections_to_ignore=None
+    ) -> dict:
+        from .._pf_client import PFClient
+
+        client = client or PFClient()
+        connection_names = SubmitterHelper.get_used_connection_names(tools_meta=tools_meta, flow_dag=flow.dag)
+        connections_to_ignore = connections_to_ignore or []
+        result = {}
+        for n in connection_names:
+            if n not in connections_to_ignore:
+                conn = client.connections.get(name=n, with_secrets=True)
+                result[n] = conn._to_execution_connection_dict()
+        return result
+
+    @staticmethod
+    def get_used_connection_names(tools_meta: dict, flow_dag: dict):
```

Comment:
Do we have a test for this?

Reply:
e2e tests will be added in cshap side given only csharp will call this function for now. A unit test is added for this.